virtual-desktop/src/app/window-manager/mvd-window-manager/launchbar/launchbar-instance-view/launchbar-instance-view.component.ts

```diff
@@ -26,8 +26,20 @@ export class LaunchbarInstanceViewComponent {
   }
 
   ngOnInit(){
-    (<HTMLImageElement>document.getElementsByClassName("instance-viewer")[0]).style.left = 
-      -(180 + (105  * (this.launchbarItem.windowPreviewsIds.length - 2))) + 'px';
+    //Doesn't check right bounds yet, for somereason bounds.right isn't giving the expected values
+    //and not sure why at this time
+    this.launchbarItem.showIconLabel = false;
+    let bounds = (<HTMLImageElement>document.getElementsByClassName("instance-viewer")[0]).getBoundingClientRect();
+    //260 offset is height of the instance viewer plus the height of the taskbar
+    (<HTMLImageElement>document.getElementsByClassName("instance-viewer")[0]).style.top = (document.body.clientHeight - 250) + 'px';
+    if (bounds != null) {
+      if (bounds.left - (180 + (105  * (this.launchbarItem.windowPreviewsIds.length - 2)))  < 0 ) {
```

Comment:
We always do constants with CAPITALS_AND_UNDERSCORES (at the top of files) to keep track of specific numbers like this, since it makes it easy to change later as well as to be more self-explanatory as to what is actually happening.